other/bootstrap_daemon/src/config.c

```diff
@@ -75,7 +79,14 @@ static void parse_tcp_relay_ports_config(config_t *cfg, uint16_t **tcp_relay_por
 
         // The loop above skips invalid ports, so we adjust the allocated memory size
         if ((*tcp_relay_port_count) > 0) {
-            *tcp_relay_ports = (uint16_t *)realloc(*tcp_relay_ports, (*tcp_relay_port_count) * sizeof(uint16_t));
+            uint16_t *tmp = (uint16_t *)realloc(*tcp_relay_ports, (*tcp_relay_port_count) * sizeof(uint16_t));
```

Comment:
I'm actually not certain about this. It seems to me that this realloc always resizes down? Why do we do this at all?